docs/GlobalRunner.md

```diff
@@ -20,6 +20,8 @@ For every specified _testing module_ `run-all.ps1` will run every specified benc
 
 The script will run benchmarks on _testing modules_ using the corresponding [_benchmark runners_](./Architecture.md#benchmark-runners). It will enforce the hard timeout (default - 5 minutes, overridable in the corresponding argument) by terminating the runner processes as necessary. It will check that the runner produces a file with timings and a file with the result Jacobian. Then it will compare the result Jacobian to the golden one and output the result of the comparison in JSON format. Unless overridden in the corresponding argument, `run-all.ps1` will delete files with Jacobians it considered correct during comparison.
 
+The script checks certain timeouts. This means that benchmark for the bigger test will not be run if the run test for the smaller size finishes with timeout. Thus, if GMM objective benchmark test is terminated due to timeout in case of `K = 10` and some `d` then tests for bigger `K` values will not be run (both for the current `d` value or for bigger `d` values), but for `1k` and `10k` such a checking is independent. For LSTM certain timeout control is the same (but for `l` and `c` instead of `d` and `K` respectively). For BA and Hand test with bigger order number will not be run if any test with less number is terminated due to timeout. So, for them tests are expected to be sorted respective their sizes.
```

Comment:
I suggest making this description more symmetrical between `d` and `K` (it's only an implementation detail that `K` is the inner loop and `d` the outer).  Perhaps something like "Thus, a GMM objective test will not be run in any case where the `d` and `K` values are both greater than or equal to the `d` and `K` values of a previously timed-out test."

Reply:
Changed